src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,1046 @@
+        if (value == null) {
+            if (logger.isDebugEnabled()) {
+                logger.debug(property + " property is not available.");
+            }
+            return null;
+        }
+        if (type.equals(String.class)) {
+            return (T) value.asText();
+        } else if (type.equals(Integer.class)) {
+            return (T) Integer.valueOf(value.asInt());
+        } else if (type.equals(Boolean.class)) {
+            return (T) Boolean.valueOf(value.asBoolean());
+        } else if (type.equals(Float.class)) {
+            return (T) Float.valueOf(value.floatValue());
+        } else if (type.equals(Long.class)) {
+            return (T) Long.valueOf(value.longValue());
+        } else if (type.equals(ArrayNode.class)) {
+            return (T) value;
+        }
+        throw new IllegalArgumentException(String.format("Unable to retrieve value for property %s and type %s", property, type.getName()));
+    }
+
+    @Override
+    protected void authenticate() throws Exception {
+        // authenticate and get sessionId like "PSN0HppfZap7wtV9MgTeGKLZL+q8q+65Te6g/r61KLqC26+thY"
+        ObjectNode request = JsonNodeFactory.instance.objectNode();
+        request.put("user", getLogin());
+        request.put("password", getPassword());
+
+        JsonNode json = doPost(SESSION, request, JsonNode.class);
+
+        Boolean success = getJsonProperty(json, "success", Boolean.class);
+        if (success == null ? false : success) {
+            sessionId = getJsonProperty(json, "sessionId", String.class);
+        } else {
+            throw new FailedLoginException("Unable to login.");
+        }
+    }
+
+    @Override
+    protected HttpHeaders putExtraRequestHeaders(HttpMethod httpMethod, String uri, HttpHeaders headers) throws Exception {
+        if (sessionId != null && !uri.equals(SESSION)) {
+            headers.add("Cookie", String.format("session_id=%s; Path=/; Domain=%s; Secure; HttpOnly;", sessionId, getHost()));
+        }
+        return super.putExtraRequestHeaders(httpMethod, uri, headers);
+    }
+
+    @Override
+    public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+        String property = controllableProperty.getProperty();
+        String value = String.valueOf(controllableProperty.getValue());
+
+        checkAndInitialize();
+        controlOperationsLock.lock();
+        try {
+            switch (property) {
+                case "Mute Microphones":
+                    if (value.equals("0")) {
+                        unmute();
+                    } else {
+                        mute();
+                    }
+                    updateLatestControlTimestamp();
+                    break;
+                case "Mute Local Video":
+                    updateVideoMuteStatus(value.equals("1"));
+                    break;
+                case "Audio Volume":
+                    updateVolumeLevel(Math.round(Float.parseFloat(value)));

```

Comment:
According to the updateVolumeLevel method documentation, it returns boolean indicating success. If that's the case, why it is not checked here for success?